rsopt/util.py

```diff
@@ -55,11 +55,14 @@ def return_nodelist(nodelist_string):
 def return_used_nodes():
     """Returns all used processor names to rank 0 or an empty list if MPI not used. For ranks != 0  returns None."""
     try:
-        from mpi4py import MPI
+        import mpi4py
```

Comment:
I think we do need knowledge still of whether we are operating in MPI or not here. That could be changed to be something that is passed as an argument. Though we would need to decide how best to inject that into `Setup` where `return_used_nodes` is used. 
Otherwise it would be best to encapsulate the check and reuse that here.

Reply:
I'd rather not call the same environment check multiple times because the Init is in there -- we need a guard around that, at the least, calling `mpi4py.MPI.Is_initialized()`, at the least, before returning a saved environment (i.e. `mpi.py:active_env`)